include/sampleflow/consumers/average_cosinus.h

```diff
@@ -168,7 +168,7 @@ namespace SampleFlow
                 }
               update = update/(std::sqrt(norm1*norm2));
               update -= current_avg_cosine[i];
-              update /= static_cast<double>(n_samples-i);
+              update /= static_cast<double>(n_samples-i-1);
```

Comment:
I'll trust you on this one :-)

Reply:
"i" starts at 0, while we minimum degrees of freedom we lose is 1. With big n, doesn't make almost no difference.